src/main/java/com/hp/application/automation/tools/run/SvExportBuilder.java

```diff
@@ -0,0 +1,143 @@
+import com.hp.sv.jsvconfigurator.processor.ExportProcessor;
+import com.hp.sv.jsvconfigurator.processor.IChmodeProcessor;
+import com.hp.sv.jsvconfigurator.serverclient.ICommandExecutor;
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.model.AbstractBuild;
+import hudson.model.BuildListener;
+import hudson.util.FormValidation;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.filefilter.DirectoryFileFilter;
+import org.apache.commons.io.filefilter.SuffixFileFilter;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+public class SvExportBuilder extends AbstractSvRunBuilder<SvExportModel> {
+
+    @DataBoundConstructor
+    public SvExportBuilder(String serverName, boolean force, String targetDirectory, boolean cleanTargetDirectory,
+                           SvServiceSelectionModel serviceSelection, boolean switchToStandByFirst) {
+        super(new SvExportModel(serverName, force, targetDirectory, cleanTargetDirectory, serviceSelection, switchToStandByFirst));
+    }
+
+    @Override
+    protected void logConfig(PrintStream logger, String prefix) {
+        logger.println(prefix + "Target Directory: " + model.getTargetDirectory());
+        logger.println(prefix + "Switch to Stand-By: " + model.isSwitchToStandByFirst());
+        super.logConfig(logger, prefix);
+    }
+
+    @Override
+    public DescriptorImpl getDescriptor() {
+        return (DescriptorImpl) super.getDescriptor();
+    }
+
+    @Override
+    public boolean performImpl(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws Exception {
+        PrintStream logger = listener.getLogger();
+
+        ExportProcessor exportProcessor = new ExportProcessor(null);
+        IChmodeProcessor chmodeProcessor = new ChmodeProcessor(null);
+
+        ICommandExecutor exec = createCommandExecutor();
+
+        verifyNotNull(model.getTargetDirectory(), "Target directory must be set");
+
+        String targetDirectory = build.getWorkspace().child(model.getTargetDirectory()).getRemote();
+
+        if (model.isCleanTargetDirectory()) {
+            cleanTargetDirectory(logger, targetDirectory);
+        }
+
+        for (ServiceInfo serviceInfo : getServiceList(false, logger, build)) {
+            if (model.isSwitchToStandByFirst()) {
+                switchToStandBy(serviceInfo, chmodeProcessor, exec, logger);
+            }
+
+            logger.printf("  Exporting service '%s' [%s] to %s %n", serviceInfo.getName(), serviceInfo.getId(), targetDirectory);
+            verifyNotLearningBeforeExport(logger, exec, serviceInfo);
+            exportProcessor.process(exec, targetDirectory, serviceInfo.getId(), false);
+        }
+
+        return true;
+    }
+
+    private void verifyNotLearningBeforeExport(PrintStream logger, ICommandExecutor exec, ServiceInfo serviceInfo)
+            throws CommunicatorException, CommandExecutorException {
```

Comment:
it might happen due to "CommunicatorException", "CommandExecutorException" both extend the same class but thrown independently and being caught by Exception. So they can be both covered by "AbstractSVCException".